compiler/fme-apply/src/EqualizePatternRead.test.cpp

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright (c) 2024 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "EqualizePatternRead.h"
+
+#include <fstream>
+#include <cstdio>
+
+#include <gtest/gtest.h>
+
+using namespace fme_apply;
+
+namespace
+{
+
+void write_to_file(const std::string &filename, const std::string &content)
+{
+  std::ofstream fs(filename, std::ofstream::binary);
+  if (fs.fail())
+    throw std::runtime_error("Cannot open file \"" + filename + "\".\n");
+  if (fs.write(content.c_str(), content.size()).fail())
+    throw std::runtime_error("Failed to write data to file \"" + filename + "\".\n");
+}
+
+class EqualizePatternReadTest : public ::testing::Test
+{
+public:
+  EqualizePatternReadTest() { _filename = "test.json"; }
+
+protected:
+  virtual void SetUp() override { std::remove(_filename.c_str()); }
+
+  virtual void TearDown() override { std::remove(_filename.c_str()); }
+
+protected:
+  std::string _filename;
+};
+
+} // namespace
+
+TEST_F(EqualizePatternReadTest, simple)
+{
+  const std::string contents = "["
+                               "{"
+                               "\"front\": \"f\","
+                               "\"back\": \"b\","
```

Comment:
(optional) It would be more helpful in case of users take it as a reference if it has node-like name instead of 'f' or 'b'